packages/backend/src/links/links.service.ts

```diff
@@ -1,45 +1,64 @@
-import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
-import { CreateLinkDto } from './dto/create-link.dto';
-import { UpdateLinkDto } from './dto/update-link.dto';
-import { PrismaService } from 'nestjs-prisma';
-import { Prisma } from '@prisma/client';
+import { BadRequestException, HttpException, Injectable, NotFoundException } from '@nestjs/common'
+import { CreateLinkDto } from './dto/create-link.dto'
+import { UpdateLinkDto } from './dto/update-link.dto'
+import { PrismaService } from 'nestjs-prisma'
+import { Prisma } from '@prisma/client'
 
 @Injectable()
 export class LinksService {
   constructor(private readonly prisma: PrismaService) {}
   async create(createLinkDto: CreateLinkDto) {
     try {
-      return await this.prisma.link.create({ data: createLinkDto });
+      if (this.checkUrl(createLinkDto.url)) {
+        return await this.prisma.link.create({ data: createLinkDto })
+      } else {
+        throw new HttpException('The url you entered is not found!', 404)
```

Comment:
I think this should be a BadRequest as well. We usually send not found when the primary entity of the controller is not found, so link in this case.